src/modinfo.c

```diff
@@ -2696,13 +2731,20 @@ sr_modinfo_module_data_load(struct sr_mod_info_s *mod_info, struct sr_mod_info_m
     if ((mod_info->ds == SR_DS_OPERATIONAL) && (mod_info->ds2 == SR_DS_OPERATIONAL)) {
         assert(sess);
 
-        /* check if we have any push data */
-        if (!sr_modinfo_sess_has_push_oper_data(sess, mod->ly_mod->name)) {
+        /* check if we have any push data (and load it if it's already in the cache) */
+        if (!sr_modinfo_module_data_cache_load(sess, mod->ly_mod->name, &mod_data)) {
             return NULL;
         }
 
-        /* load module data */
-        if ((err_info = sr_module_file_data_append(mod->ly_mod, mod->ds_handle, mod_info->ds2,
+        if (mod_data) {
+            /* duplicate the cached data for use */
+            if ((err_info = sr_lyd_dup(mod_data, NULL, LYD_DUP_RECURSIVE | LYD_DUP_WITH_FLAGS, 1, &dup))) {
+                return err_info;
+            }
+            lyd_insert_sibling(mod_info->data, dup, &mod_info->data);
+        }
+        /* load module data from the ds plugin because it is not in the cache */
+        else if ((err_info = sr_module_file_data_append(mod->ly_mod, mod->ds_handle, mod_info->ds2,
```

Comment:
Please keep it on a single line, `} else if (...`. You can put the comment right below `if`.